src/components/GpaCalculator/GpaCalculator.vue

```diff
@@ -0,0 +1,295 @@
+      var index = this.courses.indexOf(course);
+      course.hasFinal = (gradeIndex == 0);
+      this.$set(this.courses, index, course);
+      this.calculateSemesterGrades();
+    },
+    calculateSemesterGrades() {
+      var pointTotal = 0;
+      var gpa = 0.0;
+
+      for (var x = 0; x < this.courses.length; x++) { //for every course
+        var course = this.courses[x];
+        pointTotal = 0;
+        gpa = 4;
+
+        for (var y = 0; y < (course.hasFinal ? 4 : 3); y++) { //for each grades
+          pointTotal += 12 - course.grades[y];
+        }
+
+        var scoresFinal=[[48,46],[45,42],[41,38],[37,34],[33,30],[29,26],[25,22],[21,18],[17,14],[13,10],[9,6],[5,3],[2,0]];
+        var scoresNoFinal=[[36,35],[34,32],[31,29],[28,26],[25,23],[22,20],[19,17],[16,14],[13,11],[10,8],[7,5],[4,2],[1,0]];
+        var grade = "";
+
+        for (var z = 0; z < scoresNoFinal.length; z++) {
+          if (!course.hasFinal) {
+            if (
+              pointTotal >= scoresNoFinal[z][1] &&
+              pointTotal <= scoresNoFinal[z][0]
+            ) {
+              grade = this.gradeLabels[z];
+            }
+          } else {
+            if (
+              pointTotal >= scoresFinal[z][1] &&
+              pointTotal <= scoresFinal[z][0]
+            ) {
+              grade = this.gradeLabels[z];
+            }
+          }
+        }
+        if (grade.includes("A")) {
+          gpa = 4;
+        } else if (grade.includes("B")) {
+          gpa = 3;
+        } else if (grade.includes("C")) {
+          gpa = 2;
+        } else if (grade.includes("D")) {
+          gpa = 1;
+        } else {
+          gpa = 0;
+        }
+        gpa += course.selectedLevel == 1 ? 0.5 : course.selectedLevel == 2 ? 1.0 : 0;
+        course.semesterGrade = grade;
+        if (grade.includes("F")) {
+          gpa = 0.0;
+        }
+        course.gpa = gpa;
+        this.$set(this.courses, x, course);
+      }
+      var gpaSum = 0.0;
+      this.courses.forEach(function (course) {
+        gpaSum += parseFloat(course.gpa);
+      });
+
+      var average = parseFloat(gpaSum) / parseFloat(this.courses.length);
+      this.averageGpa = Number(average.toFixed(2));
+    },
+  },
+};
+</script>
+
+<style lang="sass" scoped>
+@import 'src/styles/style.sass'
+
+[contenteditable="true"]:focus
+  outline: 0px solid transparent
+
+body
+  background-color: green
+
+.add-course-button
+  width: 130px
+  margin: 0 auto
+
+.topCard
+  max-width: 976px
+  margin: 0px 12px
+  +desktop
+    margin: 0px auto !important
+
+.overall-gpa
+  margin: 4px
+  font-size: 3.5em
+  font-weight: 500
+
+.final-gpa
+  color: var(--color)
+  font-size: 2em
+  text-align: center
+
+.gradeLabel
+  padding: 0px 0px
+  margin: 0 auto
+  text-align: center
+  font-size: 3em
+
+.wrapper
+  overflow: visible !important
+
+.card
+  overflow: visible
+  margin: 0px 15px
+  background-color: white
+  border-radius: 10px
+
+.gradeSelector
+  padding: 4px
```

Comment:
can we make this `margin` instead of `padding`?
```suggestion
  margin: 4px
```

just fixes a minor style issue:
![image](https://user-images.githubusercontent.com/22779056/91129446-ac052c80-e66f-11ea-9505-62e2db1802c7.png)
to
![image](https://user-images.githubusercontent.com/22779056/91129473-b58e9480-e66f-11ea-99f8-bc9ea890e94f.png)

Reply:
please fix whatever you want, thanks :)